the-blue-alliance-ios/AppDelegate.swift

```diff
@@ -146,7 +99,60 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
 
     // MARK: Private
 
-    func setupAppearance() {
+    private func rootSplitViewController() -> UISplitViewController {
+        let mainStoryboard = UIStoryboard(name: "Main", bundle: nil)
+        guard let splitViewController = mainStoryboard.instantiateInitialViewController() as? UISplitViewController else {
+            fatalError("Unable to load root split view controller")
+        }
+        splitViewController.preferredDisplayMode = .allVisible
+        splitViewController.delegate = self
+        return splitViewController
+    }
+
+    private func launchViewController() -> UIViewController {
+        let launchStoryboard = UIStoryboard(name: "LaunchScreen", bundle: nil)
+        guard let launchViewController = launchStoryboard.instantiateInitialViewController() else {
+            fatalError("Unable to load launch view controller")
+        }
+        return launchViewController
+    }
+
+    private static func showFatalError(_ error: NSError, in window: UIWindow) {
+        let alertController = UIAlertController(title: "Error Loading Data",
+                                                message: "There was an error loading local data - try reinstalling The Blue Alliance",
+                                                preferredStyle: .alert)
+        alertController.addAction(UIAlertAction(title: "Close", style: .default, handler: { (_) in
+            fatalError("Unresolved error \(error), \(error.userInfo)")
+        }))
+        window.rootViewController?.present(alertController, animated: true, completion: nil)
+    }
+
+    private static func setupPushServiceDelegates(with pushService: PushService) {
+        Messaging.messaging().delegate = pushService
```

Comment:
We can always move these back up if they're not worth breaking out in to different spots... maybe make it slightly more readable